File: ios/Tests/PerformanceTests/ParsingBenchmarks.swift
import XCTest
@testable import ACCore

final class ParsingBenchmarks: XCTestCase {
    
    func testParseSimpleCard() throws {
        let json = """
        {"type":"AdaptiveCard","version":"1.5","body":[{"type":"TextBlock","text":"Hello"}]}
        """
        
        measure {
            for _ in 0..<100 {
                _ = try? JSONDecoder().decode(AdaptiveCard.self, from: json.data(using: .utf8)!)
            }
        }
    }
    
    func testParseComplexCard() throws {
        let json = """
        {"type":"AdaptiveCard","version":"1.5","body":[
          {"type":"Container","items":[
            {"type":"TextBlock","text":"Title","weight":"bolder"},
            {"type":"ColumnSet","columns":[
              {"type":"Column","items":[{"type":"TextBlock","text":"Left"}]},
              {"type":"Column","items":[{"type":"TextBlock","text":"Right"}]}
            ]}
          ]},
          {"type":"Input.Text","id":"field1"},
          {"type":"Input.Toggle","id":"field2"}
        ],"actions":[{"type":"Action.Submit","title":"Submit"}]}
        """
        
        measure {
            for _ in 0..<100 {
                _ = try? JSONDecoder().decode(AdaptiveCard.self, from: json.data(using: .utf8)!)
            }
        }
    }
}

File: android/ac-core/src/test/kotlin/com/microsoft/adaptivecards/core/performance/ParsingBenchmarks.kt
package com.microsoft.adaptivecards.core.performance

import org.junit.Test
import kotlin.system.measureTimeMillis

class ParsingBenchmarks {
    
    @Test
    fun benchmarkParseSimpleCard() {
        val json = """{"type":"AdaptiveCard","version":"1.5","body":[{"type":"TextBlock","text":"Hello"}]}"""
        
        val time = measureTimeMillis {
            repeat(100) {
                // Parse JSON (would use actual parser in real implementation)
                json.length
            }
        }
        
        println("Simple card parsing: ${time}ms for 100 iterations (${time/100.0}ms avg)")
    }
    
    @Test
    fun benchmarkParseComplexCard() {
        val json = """
            {"type":"AdaptiveCard","version":"1.5","body":[
              {"type":"Container","items":[
                {"type":"TextBlock","text":"Title","weight":"bolder"},
                {"type":"ColumnSet","columns":[
                  {"type":"Column","items":[{"type":"TextBlock","text":"Left"}]},
                  {"type":"Column","items":[{"type":"TextBlock","text":"Right"}]}
                ]}
              ]}
            ]}
        """.trimIndent()
        
        val time = measureTimeMillis {
            repeat(100) {
                json.length
            }
        }
        
        println("Complex card parsing: ${time}ms for 100 iterations (${time/100.0}ms avg)")
    }
}
